components/calendar.tsx

```diff
@@ -0,0 +1,469 @@
+import {
+  RiArrowLeftSLine,
+  RiArrowRightSLine,
+  RiCheckLine,
+} from "@remixicon/react";
+import {
+  HTMLAttributes,
+  CSSProperties,
+  MutableRefObject,
+  useState,
+  useEffect,
+  Ref,
+  ChangeEvent,
+} from "react";
+import { Button } from "./button";
+import { cn } from "../lib/utils";
+
+import Combobox from "./combobox";
+
+interface CalendarDrawerProps {
+  options?: string[];
+  highlightedIndex?: number;
+  setHighlightedIndex?: (index: number) => void;
+  setInputValue: (value: string) => void;
+  inputValue: string;
+  setIsOpen?: (open: boolean) => void;
+  getFloatingProps?: (
+    userProps?: HTMLAttributes<HTMLUListElement>
+  ) => HTMLAttributes<HTMLUListElement>;
+  refs?: { setFloating: Ref<HTMLUListElement> };
+  floatingStyles?: CSSProperties;
+  listRef?: MutableRefObject<(HTMLLIElement | null)[]>;
+  dayNames: string[];
+  monthNames: string[];
+  disableWeekend?: boolean;
+  minDateOffset?: number;
+  maxDateOffset?: number;
+  format?: FormatProps;
+  className?: string;
+}
+
+interface OpenBoxProps {
+  open: boolean;
+  month: string;
+  year: string;
+}
+
+type FormatProps = "mm/dd/yyyy" | "yyyy-mm-dd" | "dd/mm/yyyy";
+type DateBoxOpen = "open" | "month" | "year";
+
+function CalendarDrawer({
```

Comment:
let's call this `Calendar` and the props is `CalendarProps` ... it is not a drawer in this context